flink-end-to-end-tests/test-scripts/common.sh

```diff
@@ -456,14 +456,21 @@ function wait_job_running {
   exit 1
 }
 
+function wait_job_terminal_state_mesos {
+  local job=$1
+  local expected_terminal_state=$2
+  wait_job_terminal_state $1 $2 "mesos-appmaster"
+}
+
 function wait_job_terminal_state {
   local job=$1
   local expected_terminal_state=$2
+  local log_file_name=${3:-standalonesession}
 
   echo "Waiting for job ($job) to reach terminal state $expected_terminal_state ..."
 
   while : ; do
-    local N=$(grep -o "Job $job reached globally terminal state .*" $FLINK_DIR/log/*standalonesession*.log | tail -1 || true)
+    local N=$(grep -o "Job $job reached globally terminal state .*" $FLINK_DIR/log/*$log_file_name*.log | tail -1 || true)
```

Comment:
It's really a pity that we need a separate file pattern for the mesos appmaster but fixing that would be outside the scope of this ticket. However, I wonder if it makes sense to move the mesos flavored functions to a separate file.

Reply:
Agreed. We could move these functions to `common_mesos.sh`.